cucumber-core/src/main/java/io/cucumber/core/runner/CachingGlue.java

```diff
@@ -130,11 +136,13 @@ public void addAfterStepHook(HookDefinition hookDefinition) {
     @Override
     public void addParameterType(ParameterTypeDefinition parameterType) {
         parameterTypeDefinitions.add(parameterType);
+        dirtyCache = true;
     }
 
     @Override
     public void addDataTableType(DataTableTypeDefinition dataTableType) {
         dataTableTypeDefinitions.add(dataTableType);
+        dirtyCache = true;
```

Comment:
To wit, data tables are not involved in regex creation. So they shouldn't dirty the cache.